### PR TITLE
op-node/flags: Add easy way to deprecate flags

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This adds a deprecated flags section to warn users that the flag will be removed in a future version of the software. I had to modify the P2P Flags structure to make it work nicely.

I also added a test to ensure that deprecated flags are hidden & that all flags have at least one env var.

### CODE REVIEW

File: op-service/util.go [hunk 1]
```diff
@@ -62,6 +62,14 @@ func validateEnvVars(prefix string, providedEnvVars []string, definedEnvVars map
 	return out
 }
 
+func WarnOnDeprecatedFlags(ctx *cli.Context, deprecatedFlags []cli.Flag, log log.Logger) {
```
Comment (trianglesphere) [inline op-service/util.go:66]:
```suggestion
// WarnOnDeprecatedFlags iterates through the provided deprecatedFlags and logs a warning for each that is set.
func WarnOnDeprecatedFlags(ctx *cli.Context, deprecatedFlags []cli.Flag, log log.Logger) {
```

File: op-node/flags/p2p_flags.go [hunk 2]
```diff
@@ -55,6 +55,30 @@ var (
 	SyncReqRespName        = "p2p.sync.req-resp"
 )
 
+func DeprecatedP2PFlags(envPrefix string) []cli.Flag {
```
Comment (sebastianst) [inline op-node/flags/p2p_flags.go]: Why does this func need to be public?

**Review by sebastianst — COMMENTED** [review]
LGTM, one q

File: op-node/flags/p2p_flags.go (hunk 2 above)
Comment (trianglesphere) [reply op-node/flags/p2p_flags.go]: oh, p2pdeprecated doesn't. Deprecated does

**Review by sebastianst — APPROVED** [review]

**Pull request merged**